api/src/org/apache/cloudstack/api/command/admin/template/RegisterTemplateCmdByAdmin.java

```diff
@@ -40,10 +40,23 @@
     @Override
     public void execute() throws ResourceAllocationException{
         try {
+            if ((zoneId != null) && (zoneIds != null && !zoneIds.isEmpty()))
```

Comment:
Again, extra brackets are for increase in readability. There are two things being checked here, the ZoneID shouldnt be null and, if zoneIds list shouldn't be empty.